### PR TITLE
fix: update Jumpstart errors to EvalAlgorithmInternalError

### PR DESCRIPTION
**Pull request by keerthanvasist**
*Description of changes:*
fix: update Jumpstart errors to EvalAlgorithmInternalError since they are not client errors but errors in Jumpstart or fmeval 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/fmeval/model_runners/extractors/jumpstart_extractor.py [hunk 1]
```diff
@@ -56,15 +56,19 @@ def __init__(
...
+            output_jmespath_expressions = compile_jmespath(GENERATED_TEXT_JMESPATH_EXPRESSION).search(
+                model_spec_key[DEFAULT_PAYLOADS]
+            )
+            util.assert_condition(
+                output_jmespath_expressions,
+                f"Jumpstart model {jumpstart_model_id} is likely not supported. If you know it is generates text, "
+                f"please provide output and log_probability jmespaths to the JumpStartModelRunner",
+            )
+            self._output_jmespath_compiler = compile_jmespath(output_jmespath_expressions[0])
+        except EvalAlgorithmClientError as e:
```
Comment (pinaraws) [inline src/fmeval/model_runners/extractors/jumpstart_extractor.py:70]: does this come from assert_condition? Let's throw internal error directly instead of mutating it.

File: src/fmeval/model_runners/extractors/jumpstart_extractor.py [hunk 2]
```diff
@@ -56,15 +56,19 @@ def __init__(
             self._sagemaker_session.boto_region_name,
         )
         util.require(DEFAULT_PAYLOADS in model_spec_key, f"Model: {jumpstart_model_id} is not supported at this time")
-        output_jmespath_expressions = compile_jmespath(GENERATED_TEXT_JMESPATH_EXPRESSION).search(
-            model_spec_key[DEFAULT_PAYLOADS]
-        )
-        util.require(
-            output_jmespath_expressions,
-            f"Jumpstart model {jumpstart_model_id} is likely not supported. If you know it is generates text, "
```
Comment (pinaraws) [inline src/fmeval/model_runners/extractors/jumpstart_extractor.py:64]: Let's reword this to something like: "Default payload for Jumpstart could not be constructed. Please configure output and log probability instead."

File: src/fmeval/model_runners/extractors/jumpstart_extractor.py (hunk 1 above)
Comment (keerthanvasist) [reply src/fmeval/model_runners/extractors/jumpstart_extractor.py:70]: `assert_condition` throws `EvalAlgorithmInternalError`. The error we are catching `EvalAlgorithmClientError` comes from `compile_jmespath` - this is used in other places where it is indeed a client error.

File: src/fmeval/model_runners/extractors/jumpstart_extractor.py [hunk 3]
```diff
@@ -55,16 +57,31 @@ def __init__(
...
+            DEFAULT_PAYLOADS in model_spec_key, f"JumpStart Model: {jumpstart_model_id} is not supported at this time"
+        )
+
+        try:
+            output_jmespath_expressions = jmespath.compile(GENERATED_TEXT_JMESPATH_EXPRESSION).search(
+                model_spec_key[DEFAULT_PAYLOADS]
+            )
+        except (TypeError, JMESPathError) as e:
+            raise EvalAlgorithmInternalError(
+                f"Unable find the generated_text key in the default payload for JumpStart model: {jumpstart_model_id}"
```
Comment (pinaraws) [inline src/fmeval/model_runners/extractors/jumpstart_extractor.py:70]: nit: add period and space at the end `. "`

**Review by pinaraws — DISMISSED** [review]

**Review by Satish615 — APPROVED** [review]
